### ISSUE TITLE
Latest SpringBoot quickstarter broken

### ISSUE THREAD

**Issue by georgfedermann** [opendevstack/ods-quickstarters#418]
**Describe the bug**
When creating a spring boot component using the quickstarter, the build fails with the error message below.

**Log Output (ensure to remove any confidential information like tokens, project names, etc.**
```
Step 1/6 : FROM adoptopenjdk/openjdk11:alpine-jre
 ---> d12bace74cab
Step 2/6 : COPY app.jar app.jar
error: build error: COPY failed: stat /var/lib/docker/tmp/docker-builder799079037/app.jar: no such file or directory
```

build.gradle
```
plugins {
	id 'org.springframework.boot' version '2.3.2.RELEASE'
	id 'io.spring.dependency-management' version '1.0.9.RELEASE'
	id 'org.asciidoctor.convert' version '1.5.8'
	id 'java'
}

group = 'unitt'
version = '0.0.1-SNAPSHOT'
sourceCompatibility = '11'

configurations {
	compileOnly {
		extendsFrom annotationProcessor
	}
}

repositories {
	mavenCentral()
}

ext {
	set('snippetsDir', file("build/generated-snippets"))
}

dependencies {
	implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
	implementation 'org.springframework.boot:spring-boot-starter-data-rest'
	implementation 'org.springframework.boot:spring-boot-starter-jersey'
	implementation 'org.springframework.boot:spring-boot-starter-security'
	implementation 'org.springframework.boot:spring-boot-starter-web'
	compileOnly 'org.projectlombok:lombok'
	runtimeOnly 'com.h2database:h2'
	annotationProcessor 'org.projectlombok:lombok'
	testImplementation('org.springframework.boot:spring-boot-starter-test') {
		exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
	}
	testImplementation 'org.springframework.restdocs:spring-restdocs-mockmvc'
	testImplementation 'org.springframework.security:spring-security-test'
}

test {
	outputs.dir snippetsDir
	useJUnitPlatform()
}

asciidoctor {
	inputs.dir snippetsDir
	dependsOn test
}
```

**Comment by clemensutschig** [comment]
Nexus and all other pieces are missing!!!!

**Comment by clemensutschig** [comment]
This is a Blocker for ods3

**Comment by renedupont** [comment]
@georgfedermann 
weird... nothing changed in a while and when I tested it yesterday it worked. How was your setup, did you use master branch etc.?
In your gradle it says spring boot 2.3.2, but in the quickstarter it uses 2.2.6 (https://github.com/opendevstack/ods-quickstarters/blob/master/be-java-springboot/Jenkinsfile#L7) 
Are you sure the gradle file was not changed?

**Comment by clemensutschig** [comment]
Latest master.. no changes to build gradle

**Comment by renedupont** [comment]
I have an idea... spring.io might have changed their api and replaced platformVersion https://github.com/opendevstack/ods-quickstarters/blob/master/be-java-springboot/Jenkinsfile#L19 and therefore it does not pick up the 2.2.6 version anymore and just picks the latest. but why the gradle build file looks like this... I'm not sure yet, will investigate later

**Comment by renedupont** [comment]
Ok my idea turned out to be wrong... I tested it again and still got no issues. It also took spring version 2.2.6

**Comment by georgfedermann** [comment]
Jenkins log from prov-springboot build

```
OpenShift Build unitt-cd/prov-be-java-springboot-unitt-master-3 from http://bitbucket.odsbox.lan:7990/scm/opendevstack/ods-quickstarters.git
Checking out git http://bitbucket.odsbox.lan:7990/scm/opendevstack/ods-quickstarters.git into /var/lib/jenkins/jobs/unitt-cd/jobs/unitt-cd-prov-be-java-springboot-unitt-master/workspace@script to read be-java-springboot/Jenkinsfile
using credential unitt-cd-cd-user-with-password
 > git rev-parse --is-inside-work-tree # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url http://bitbucket.odsbox.lan:7990/scm/opendevstack/ods-quickstarters.git # timeout=10
Fetching upstream changes from http://bitbucket.odsbox.lan:7990/scm/opendevstack/ods-quickstarters.git
 > git --version # timeout=10
using GIT_ASKPASS to set credentials unitt-cd-cd-user-with-password
 > git fetch --tags --progress http://bitbucket.odsbox.lan:7990/scm/opendevstack/ods-quickstarters.git +refs/heads/*:refs/remotes/origin/*
 > git rev-parse origin/master^{commit} # timeout=10
Checking out Revision 0a24b50f848bb7dd63c211ac374adbde9660cc05 (origin/master)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 0a24b50f848bb7dd63c211ac374adbde9660cc05
Commit message: "Fix failing test for git branches (#417)"
 > git rev-list --no-walk 0a24b50f848bb7dd63c211ac374adbde9660cc05 # timeout=10
Running in Durability level: MAX_SURVIVABILITY
[Pipeline] Start of Pipeline
[Pipeline] library
Loading library ods-jenkins-shared-library@master
Attempting to resolve master from remote references...
 > git --version # timeout=10
using GIT_ASKPASS to set credentials unitt-cd-cd-user-with-password
 > git ls-remote -h http://bitbucket.odsbox.lan:7990/scm/opendevstack/ods-jenkins-shared-library.git # timeout=10
Found match: refs/heads/master revision f0e38bdc024a6fd200d64dfc172f541f85274e1e
using credential unitt-cd-cd-user-with-password
 > git rev-parse --is-inside-work-tree # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url http://bitbucket.odsbox.lan:7990/scm/opendevstack/ods-jenkins-shared-library.git # timeout=10
Fetching without tags
Fetching upstream changes from http://bitbucket.odsbox.lan:7990/scm/opendevstack/ods-jenkins-shared-library.git
 > git --version # timeout=10
using GIT_ASKPASS to set credentials unitt-cd-cd-user-with-password
 > git fetch --no-tags --progress http://bitbucket.odsbox.lan:7990/scm/opendevstack/ods-jenkins-shared-library.git +refs/heads/*:refs/remotes/origin/*
Checking out Revision f0e38bdc024a6fd200d64dfc172f541f85274e1e (master)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f f0e38bdc024a6fd200d64dfc172f541f85274e1e
Commit message: "Merge pull request #420 from BIX-Digital/feature/general-fork"
 > git rev-list --no-walk f0e38bdc024a6fd200d64dfc172f541f85274e1e # timeout=10
[Pipeline] node
Running on Jenkins in /var/lib/jenkins/jobs/unitt-cd/jobs/unitt-cd-prov-be-java-springboot-unitt-master/workspace
[Pipeline] {
[Pipeline] }
[Pipeline] // node
[Pipeline] podTemplate
[Pipeline] {
[Pipeline] node
Still waiting to schedule task
‘Jenkins’ doesn’t have label ‘qs-a3c05e19-3bad-444d-99c4-e95b1c3f4f07’
Agent qs-a3c05e19-3bad-444d-99c4-e95b1c3f4f07-8j3g4-jxjd9 is provisioned from template Kubernetes Pod Template
---
apiVersion: "v1"
kind: "Pod"
metadata:
  annotations:
    buildUrl: "http://jenkins.unitt-cd.svc:80/job/unitt-cd/job/unitt-cd-prov-be-java-springboot-unitt-master/3/"
  labels:
    jenkins: "slave"
    jenkins/qs-a3c05e19-3bad-444d-99c4-e95b1c3f4f07: "true"
  name: "qs-a3c05e19-3bad-444d-99c4-e95b1c3f4f07-8j3g4-jxjd9"
spec:
  containers:
  - env:
    - name: "JENKINS_SECRET"
      value: "********"
    - name: "JENKINS_TUNNEL"
      value: "jenkins-jnlp.unitt-cd.svc:50000"
    - name: "JENKINS_AGENT_NAME"
      value: "qs-a3c05e19-3bad-444d-99c4-e95b1c3f4f07-8j3g4-jxjd9"
    - name: "JENKINS_NAME"
      value: "qs-a3c05e19-3bad-444d-99c4-e95b1c3f4f07-8j3g4-jxjd9"
    - name: "JENKINS_AGENT_WORKDIR"
      value: "/tmp"
    - name: "JENKINS_URL"
      value: "http://jenkins.unitt-cd.svc:80/"
    - name: "HOME"
      value: "/home/jenkins"
    image: "docker-registry.default.svc:5000/ods/jenkins-agent-maven:latest"
    imagePullPolicy: "Always"
    name: "jnlp"
    resources:
      limits:
        memory: "1Gi"
        cpu: "1"
      requests:
        memory: "512Mi"
        cpu: "100m"
    securityContext:
      privileged: false
    tty: false
    volumeMounts:
    - mountPath: "/tmp"
      name: "workspace-volume"
      readOnly: false
    workingDir: "/tmp"
  nodeSelector: {}
  restartPolicy: "Never"
  serviceAccount: "jenkins"
  volumes:
  - emptyDir: {}
    name: "workspace-volume"

Running on qs-a3c05e19-3bad-444d-99c4-e95b1c3f4f07-8j3g4-jxjd9 in /tmp/workspace/unitt-cd/unitt-cd-prov-be-java-springboot-unitt-master
[Pipeline] {
[Pipeline] echo
[Checkout quickstarter-springboot] **** STARTING stage 'Checkout quickstarter' for component 'springboot' branch 'master' ****
[Pipeline] stage
[Pipeline] { (Checkout quickstarter)
[Pipeline] checkout
using credential unitt-cd-cd-user-with-password
Cloning the remote Git repository
Cloning repository http://bitbucket.odsbox.lan:7990/scm/opendevstack/ods-quickstarters.git
 > git init /tmp/workspace/unitt-cd/unitt-cd-prov-be-java-springboot-unitt-master # timeout=10
Fetching upstream changes from http://bitbucket.odsbox.lan:7990/scm/opendevstack/ods-quickstarters.git
 > git --version # timeout=10
using GIT_ASKPASS to set credentials unitt-cd-cd-user-with-password
 > git fetch --tags --progress http://bitbucket.odsbox.lan:7990/scm/opendevstack/ods-quickstarters.git +refs/heads/*:refs/remotes/origin/*
Checking out Revision 0a24b50f848bb7dd63c211ac374adbde9660cc05 (origin/master)
 > git config remote.origin.url http://bitbucket.odsbox.lan:7990/scm/opendevstack/ods-quickstarters.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url http://bitbucket.odsbox.lan:7990/scm/opendevstack/ods-quickstarters.git # timeout=10
Fetching upstream changes from http://bitbucket.odsbox.lan:7990/scm/opendevstack/ods-quickstarters.git
using GIT_ASKPASS to set credentials unitt-cd-cd-user-with-password
 > git fetch --tags --progress http://bitbucket.odsbox.lan:7990/scm/opendevstack/ods-quickstarters.git +refs/heads/*:refs/remotes/origin/*
 > git rev-parse origin/master^{commit} # timeout=10
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 0a24b50f848bb7dd63c211ac374adbde9660cc05
Commit message: "Fix failing test for git branches (#417)"
[Pipeline] }
[Pipeline] // stage
[Pipeline] echo
[Checkout quickstarter-springboot] **** ENDED stage 'Checkout quickstarter' for component 'springboot' branch 'master' **** (took 11362 ms)
[Pipeline] echo
[Initialize output directory-springboot] **** STARTING stage 'Initialize output directory' for component 'springboot' branch 'master' ****
[Pipeline] stage
[Pipeline] { (Initialize output directory)
[Pipeline] fileExists
[Pipeline] sh (Create directory 'springboot')
+ mkdir -p springboot
[Pipeline] }
[Pipeline] // stage
[Pipeline] echo
[Initialize output directory-springboot] **** ENDED stage 'Initialize output directory' for component 'springboot' branch 'master' **** (took 1823 ms)
[Pipeline] stage
[Pipeline] { (Build spring project)
[Pipeline] dir
Running in /tmp/workspace/unitt-cd/unitt-cd-prov-be-java-springboot-unitt-master/springboot
[Pipeline] {
[Pipeline] sh
+ curl https://start.spring.io/starter.tgz -d type=gradle-project -d language=java -d dependencies=web,jersey,data-jpa,h2,lombok,data-rest,restdocs,security -d platformVersion=2.2.6.RELEASE -d packaging=jar -d jvmVersion=1.11 -d groupId=unitt -d artifactId=springboot -d name=springboot -d description=OpenDevStack%20Demo%20Project%20for%20Spring%20Boot -d packageName=null
+ tar -zxvf -
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0.gitignore
gradlew.bat
HELP.md
gradlew
settings.gradle
src/
src/main/
src/main/resources/
src/main/resources/templates/
src/main/resources/application.properties
src/main/resources/static/
src/main/java/
src/main/java/com/
src/main/java/com/example/
src/main/java/com/example/demo/
src/main/java/com/example/demo/SpringbootApplication.java
src/test/
src/test/java/
src/test/java/com/
src/test/java/com/example/
src/test/java/com/example/demo/
src/test/java/com/example/demo/SpringbootApplicationTests.java
build.gradle
gradle/
gradle/wrapper/
gradle/wrapper/gradle-wrapper.properties
gradle/wrapper/gradle-wrapper.jar

100 59252  100 58955  100   297  90854    457 --:--:-- --:--:-- --:--:-- 90839
[Pipeline] withEnv
[Pipeline] {
[Pipeline] sh
+ sh /tmp/workspace/unitt-cd/unitt-cd-prov-be-java-springboot-unitt-master/be-java-springboot/postProcessProjectsettings.sh
configure in memory database in spring application.properties
/tmp/workspace/unitt-cd/unitt-cd-prov-be-java-springboot-unitt-master/be-java-springboot/postProcessProjectsettings.sh: line 6: src/main/resources/application.properties: No such file or directory
/tmp/workspace/unitt-cd/unitt-cd-prov-be-java-springboot-unitt-master/be-java-springboot/postProcessProjectsettings.sh: line 7: src/main/resources/application-dev.properties: No such file or directory
customise build.gradle - getting version
grep: gradle/wrapper/gradle-wrapper.properties: No such file or directory
gradle version: 
using /tmp/workspace/unitt-cd/unitt-cd-prov-be-java-springboot-unitt-master/be-java-springboot/templates/build-.gradle
do not add legacy nexus upload script to build.gradle
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
[Pipeline] // dir
[Pipeline] }
[Pipeline] // stage
[Pipeline] echo
[Copy files from quickstarter-springboot] **** STARTING stage 'Copy files from quickstarter' for component 'springboot' branch 'master' ****
[Pipeline] stage
[Pipeline] { (Copy files from quickstarter)
[Pipeline] sh (Copy files from 'be-java-springboot/files' to 'springboot')
+ cp -rv be-java-springboot/files/. springboot
'be-java-springboot/files/./docker' -> 'springboot/./docker'
'be-java-springboot/files/./docker/Dockerfile' -> 'springboot/./docker/Dockerfile'
'be-java-springboot/files/./metadata.yml' -> 'springboot/./metadata.yml'
'be-java-springboot/files/./release-manager.yml' -> 'springboot/./release-manager.yml'
[Pipeline] }
[Pipeline] // stage
[Pipeline] echo
[Copy files from quickstarter-springboot] **** ENDED stage 'Copy files from quickstarter' for component 'springboot' branch 'master' **** (took 379 ms)
[Pipeline] echo
[Create OpenShift resources-springboot] **** STARTING stage 'Create OpenShift resources' for component 'springboot' branch 'master' ****
[Pipeline] stage
[Pipeline] { (Create OpenShift resources)
[Pipeline] fileExists
[Pipeline] dir
Running in /tmp/workspace/unitt-cd/unitt-cd-prov-be-java-springboot-unitt-master/common/ocp-config/component-environment
[Pipeline] {
[Pipeline] sh (Create component springboot in namespace unitt-dev)
+ tailor --non-interactive -n unitt-dev apply --upsert-only --ignore-unknown-parameters --selector app=unitt-springboot --param=PROJECT=unitt --param=COMPONENT=springboot --param=ENV=dev --param=DOCKER_REGISTRY=docker-registry.default.svc:5000 --param-file /tmp/workspace/unitt-cd/unitt-cd-prov-be-java-springboot-unitt-master/be-java-springboot/ocp.env
Comparing templates in . with OCP namespace unitt-dev.
Limiting resources to svc,dc,is,bc with selector app=unitt-springboot.
Found 4 resources in OCP cluster (current state) and 4 resources in processed templates (desired state).

* svc/springboot is in sync
* is/springboot is in sync
* bc/springboot is in sync
* dc/springboot is in sync

Summary: 4 in sync, 0 to create, 0 to update, 0 to delete

[Pipeline] }
[Pipeline] // dir
[Pipeline] fileExists
[Pipeline] dir
Running in /tmp/workspace/unitt-cd/unitt-cd-prov-be-java-springboot-unitt-master/common/ocp-config/component-environment
[Pipeline] {
[Pipeline] sh (Create component springboot in namespace unitt-test)
+ tailor --non-interactive -n unitt-test apply --upsert-only --ignore-unknown-parameters --selector app=unitt-springboot --param=PROJECT=unitt --param=COMPONENT=springboot --param=ENV=test --param=DOCKER_REGISTRY=docker-registry.default.svc:5000 --param-file /tmp/workspace/unitt-cd/unitt-cd-prov-be-java-springboot-unitt-master/be-java-springboot/ocp.env
Comparing templates in . with OCP namespace unitt-test.
Limiting resources to svc,dc,is,bc with selector app=unitt-springboot.
Found 4 resources in OCP cluster (current state) and 4 resources in processed templates (desired state).

* svc/springboot is in sync
* is/springboot is in sync
* bc/springboot is in sync
* dc/springboot is in sync

Summary: 4 in sync, 0 to create, 0 to update, 0 to delete

[Pipeline] }
[Pipeline] // dir
[Pipeline] }
[Pipeline] // stage
[Pipeline] echo
[Create OpenShift resources-springboot] **** ENDED stage 'Create OpenShift resources' for component 'springboot' branch 'master' **** (took 5449 ms)
[Pipeline] echo
[Create Jenkinsfile-springboot] **** STARTING stage 'Create Jenkinsfile' for component 'springboot' branch 'master' ****
[Pipeline] stage
[Pipeline] { (Create Jenkinsfile)
[Pipeline] sh (Render 'Jenkinsfile.template' to 'Jenkinsfile')
+ sed 's|@project_id@|unitt|g; s|@component_id@|springboot|g; s|@component_type@|be-java-springboot|g; s|@git_url_http@|http://bitbucket.odsbox.lan:7990/scm/unitt/springboot.git|g; s|@ods_namespace@|ods|g; s|@ods_image_tag@|latest|g; s|@ods_git_ref@|master|g' be-java-springboot/Jenkinsfile.template
[Pipeline] }
[Pipeline] // stage
[Pipeline] echo
[Create Jenkinsfile-springboot] **** ENDED stage 'Create Jenkinsfile' for component 'springboot' branch 'master' **** (took 391 ms)
[Pipeline] echo
[Create sonar-project.properties-springboot] **** STARTING stage 'Create sonar-project.properties' for component 'springboot' branch 'master' ****
[Pipeline] stage
[Pipeline] { (Create sonar-project.properties)
[Pipeline] sh (Render 'sonar-project.properties.template' to 'sonar-project.properties')
+ sed 's|@project_id@|unitt|g; s|@component_id@|springboot|g' be-java-springboot/sonar-project.properties.template
[Pipeline] }
[Pipeline] // stage
[Pipeline] echo
[Create sonar-project.properties-springboot] **** ENDED stage 'Create sonar-project.properties' for component 'springboot' branch 'master' **** (took 350 ms)
[Pipeline] echo
[Push to remote-springboot] **** STARTING stage 'Push to remote' for component 'springboot' branch 'master' ****
[Pipeline] stage
[Pipeline] { (Push to remote)
[Pipeline] withCredentials
Masking supported pattern matches of $user or $pass
[Pipeline] {
[Pipeline] writeFile
[Pipeline] echo
wrote /home/jenkins/.git-credentials-input with credentials for **** on http://bitbucket.odsbox.lan:7990
[Pipeline] sh (setup git credential helper)
+ git config --global credential.helper store
[Pipeline] sh (setup git credential store)
+ cat /home/jenkins/.git-credentials-input
+ git credential-store store
[Pipeline] }
[Pipeline] // withCredentials
[Pipeline] dir
Running in /tmp/workspace/unitt-cd/unitt-cd-prov-be-java-springboot-unitt-master/springboot
[Pipeline] {
[Pipeline] fileExists
[Pipeline] echo
Initializing quickstarter git repo springboot @http://bitbucket.odsbox.lan:7990/scm/unitt/springboot.git
[Pipeline] sh (Copy and commit quickstarter files)
+ git clone http://bitbucket.odsbox.lan:7990/scm/unitt/springboot.git
Cloning into 'springboot'...
warning: You appear to have cloned an empty repository.
++ basename http://bitbucket.odsbox.lan:7990/scm/unitt/springboot.git
+ basename=springboot.git
+ clonedGitFolderName=springboot
++ ls springboot
++ grep -v '^l'
++ wc -l
+ files=0
+ '[' 0 == 0 ']'
++ pwd
+ echo 'Init project from scratch /tmp/workspace/unitt-cd/unitt-cd-prov-be-java-springboot-unitt-master/springboot'
Init project from scratch /tmp/workspace/unitt-cd/unitt-cd-prov-be-java-springboot-unitt-master/springboot
+ git init .
Initialized empty Git repository in /tmp/workspace/unitt-cd/unitt-cd-prov-be-java-springboot-unitt-master/springboot/.git/
+ git remote add origin http://bitbucket.odsbox.lan:7990/scm/unitt/springboot.git
+ rm -rf springboot
+ git config user.email undefined
+ git config user.name 'ODS System User'
+ git add --all .
+ git commit -m 'Initial OpenDevStack commit'
[master (root-commit) ef001e3] Initial OpenDevStack commit
 15 files changed, 473 insertions(+)
 create mode 100644 .gitignore
 create mode 100644 Jenkinsfile
 create mode 100644 build.gradle
 create mode 100644 docker/Dockerfile
 create mode 100644 gradle/wrapper/gradle-wrapper.jar
 create mode 100644 gradle/wrapper/gradle-wrapper.properties
 create mode 100755 gradlew
 create mode 100644 gradlew.bat
 create mode 100644 metadata.yml
 create mode 100644 release-manager.yml
 create mode 100644 settings.gradle
 create mode 100644 sonar-project.properties
 create mode 100644 src/main/java/com/example/demo/SpringbootApplication.java
 create mode 100644 src/main/resources/application.properties
 create mode 100644 src/test/java/com/example/demo/SpringbootApplicationTests.java
[Pipeline] echo
Pushing quickstarter git repo to http://bitbucket.odsbox.lan:7990/scm/unitt/springboot.git
[Pipeline] sh (Push to remote)
++ git rev-parse --abbrev-ref HEAD
+ git push -u origin master
To http://bitbucket.odsbox.lan:7990/scm/unitt/springboot.git
 * [new branch]      master -> master
Branch master set up to track remote branch master from origin.
[Pipeline] }
[Pipeline] // dir
[Pipeline] }
[Pipeline] // stage
[Pipeline] echo
[Push to remote-springboot] **** ENDED stage 'Push to remote' for component 'springboot' branch 'master' **** (took 1948 ms)
[Pipeline] }
[Pipeline] // node
[Pipeline] }
[Pipeline] // podTemplate
[Pipeline] End of Pipeline
Finished: SUCCESS
```

**Comment by georgfedermann** [comment]
superseded by https://github.com/opendevstack/ods-jenkins-shared-library/issues/428
